### PR TITLE
feat(url): support ref as state

### LINKED ISSUE #429: [Url] Support `MaybeRef` on `useUrlQuerySync`
Currently, `useUrlQuerySync` only accept reactive object as a state, but would be cool if we can pass in `MaybeRefOrGetter`.

### PR DESCRIPTION
**Pull request by brc-dd**
closes #429

### CODE REVIEW

File: lib/composables/Url.ts
Comment (kiaking) [inline lib/composables/Url.ts:48]: Just curious, probably I'm missing something but can't we use `toValue` here?

Comment (brc-dd) [reply lib/composables/Url.ts:48]: Yeah I was thinking that too. I'm not sure if assignments work with it. I'll take a look.

Comment (brc-dd) [reply lib/composables/Url.ts]: I initially thought assignment might not work with that. Seems to be working fine. Updated it.

Comment (kiaking) [reply lib/composables/Url.ts]: Nice. Yeah I was curious too so went ahead and checked the code but this ternary is what exactly Vue is doing 👍 
https://github.com/vuejs/core/blob/main/packages/reactivity/src/ref.ts

```ts
export function toValue<T>(source: MaybeRefOrGetter<T> | ComputedRef<T>): T {
  return isFunction(source) ? source() : unref(source)
}

export function unref<T>(ref: MaybeRef<T> | ComputedRef<T>): T {
  return isRef(ref) ? ref.value : ref
}
```

Comment (brc-dd) [reply lib/composables/Url.ts]: Yeah. I was just looking at that too 😅 We don't need toValue either. unref should work fine here and avoid extra checks for function.

Comment (kiaking) [reply lib/composables/Url.ts]: Very good point.

**Review by kiaking — APPROVED** [review]
@brc-dd 
LGTM! Please feel free to merge when you are ready, once merged, could you make an another release? 🙏

**Comment by brc-dd** [comment]
released

**Comment by kiaking** [comment]
Thanks!

**Pull request merged**
